test/template.js

```diff
@@ -0,0 +1,61 @@
+/*globals describe, it*/
+const assert = require('assert');
+const TemplateGenerator = require('../lib/cmds/template_generator');
+
+describe('TemplateGenerator', function () {
+  describe('getExternalProject', function () {
+    let templateGenerator;
+
+    before(() => {
+      templateGenerator = new TemplateGenerator();
+    });
+
+    describe('with github link', function () {
+
+      it('return correct zip filename for https link', function (done) {
+        let result = templateGenerator.getExternalProject("https://github.com/embark-framework/embark");
+        assert.deepEqual(result.url, "https://github.com/embark-framework/embark/archive/master.zip");
+        assert.deepEqual(result.filePath, ".embark/templates/embark-framework/embark/archive.zip");
+        done();
```

Comment:
Since it's a sync function, you can remove the done (from here and from the params)